fabtests/include/windows/pthread.h

```diff
@@ -0,0 +1,310 @@
+	return 0;
+}
+
+/*
+ * TODO: temporary solution
+ * Need to re-implement
+ */
+typedef void(*pthread_cleanup_callback_t)(void *);
+typedef struct pthread_cleanup_t
+{
+	pthread_cleanup_callback_t routine;
+	void *arg;
+} pthread_cleanup_t;
+
+/* Read-Write lock implementation */
+
+typedef struct {
+    SRWLOCK	lock; /* Windows Slim Reader Writer Lock */
+    bool	write_mode;
+} pthread_rwlock_t;
+typedef void pthread_rwlockattr_t;
+
+static inline int pthread_rwlock_init(pthread_rwlock_t *rwlock, const pthread_rwlockattr_t *attr)
+{
+	(void)attr;
+	if (rwlock) {
+		InitializeSRWLock(&(rwlock->lock));
+		rwlock->write_mode = false;
+		return 0;
+	}
+	return 1;
+}
+
+static inline int pthread_rwlock_destroy(pthread_rwlock_t *rwlock)
+{
+	/* No SRWLock cleanup function */
+	(void)rwlock;
+	return 0;
+}
+
+static inline int pthread_rwlock_rdlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock) {
+		AcquireSRWLockShared(&(rwlock->lock));
+		return 0;
+	}
+	return 1;
+}
+
+static inline int pthread_rwlock_tryrdlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock && TryAcquireSRWLockShared(&(rwlock->lock))) {
+		return 0;
+	}
+	return 1;
+}
+
+static inline int pthread_rwlock_wrlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock) {
+		AcquireSRWLockExclusive(&(rwlock->lock));
+		rwlock->write_mode = true;
+		return 0;
+	}
+	return 1;
+}
+
+static inline int pthread_rwlock_trywrlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock && TryAcquireSRWLockExclusive(&(rwlock->lock))) {
+		rwlock->write_mode = true;
+		return 0;
+	}
+	return 1;
+}
+
+
+static inline int pthread_rwlock_unlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock) {
+		if (rwlock->write_mode) {
+			rwlock->write_mode = false;
+			ReleaseSRWLockExclusive(&(rwlock->lock));
+		} else {
+			ReleaseSRWLockShared(&(rwlock->lock));
+		}
+		return 0;
+	}
+	return 1;
+}
+
+
+typedef struct {
+	int needed;
+	int called;
+	pthread_mutex_t mutex;
+	pthread_cond_t cond;
+} barrier_t;
+
+static inline int barrier_init(barrier_t *barrier,int needed)
+{
+    barrier->needed = needed;
+    barrier->called = 0;
+    pthread_mutex_init(&barrier->mutex, NULL);
+    pthread_cond_init(&barrier->cond, NULL);
+    return 0;
+}
+
+static inline int barrier_destroy(barrier_t *barrier)
+{
+    pthread_mutex_destroy(&barrier->mutex);
+    pthread_cond_destroy(&barrier->cond);
+    return 0;
+}
+
+static inline int barrier_wait(barrier_t *barrier)
+{
+    pthread_mutex_lock(&barrier->mutex);
+    barrier->called++;
+    if (barrier->called == barrier->needed) {
+        barrier->called = 0;
+        pthread_cond_broadcast(&barrier->cond);
+    } else {
+        pthread_cond_wait(&barrier->cond, &barrier->mutex);
+    }
+    pthread_mutex_unlock(&barrier->mutex);
+    return 0;
+}
+
+#define pthread_barrier_t barrier_t
+#define pthread_barrier_attr_t barrier_attr_t
+#define pthread_barrier_init(b,a,n) barrier_init(b,n)
+#define pthread_barrier_destroy(b) barrier_destroy(b)
+#define pthread_barrier_wait(b) barrier_wait(b)
```

Comment:
Just use the pthread_xxx names in the original function definition. No need to redefine here. Can do the same for the osx version.